### ISSUE TITLE
Very high CPU use whilst plotting with nbagg backend

### ISSUE THREAD

**Issue by thomasaarholt** [hyperspy/hyperspy#1577]
My CPU goes to 100% when plotting with the backend. This is not the case when I plot "normal" matplotlib signals.

```python
%matplotlib nbagg #or %matplotlib notebook
import hyperspy.api as hs
hs.datasets.example_signals.EDS_SEM_Spectrum().plot()
``` 
is enough to replicate the problem.

I suspect this is a hs issue rather than a jupyter issue, but I'm unsure how to check.
Interactive matplotlib plots do not show this bug.

**Comment by thomasaarholt** [comment]
I'm pretty sure that there's a while loop or something similar that keeps doing something even when the plot is not being changed.

**Comment by francisco-dlp** [comment]
That's probably the trick that nbagg uses to achieve interactivity. I don't think that there is anything that we can do about it other than using e.g. bokeh for interactive plotting in the notebook.

**Comment by thomasaarholt** [comment]
Is this an issue on Linux as well?

The thing is, the example above is a "static" plot (not a SI I mean), yet 
```python
plt.plot([1,2,3])
```
shows no CPU use.

**Comment by thomasaarholt** [comment]
Actually, this example does NOT show a high CPU use:

[Interactive_Example.ipynb.zip](https://github.com/hyperspy/hyperspy/files/908120/Interactive_Example.ipynb.zip)

**Comment by francisco-dlp** [comment]
Interesting. Then, It's worth investigating which of the mpl features that we use causes the issue.

**Comment by francisco-dlp** [comment]
I have tried to fix it in #1544, could you confirm that it fixes the issue for you?

**Comment by thomasaarholt** [comment]
~~I just tried, and it does unfortunately not fix it.~~

**Comment by francisco-dlp** [comment]
Are you sure that you were testing the right branch? The improvement in huge in my linux desktop and I can't see why it wouldn't be so in a Mac.

**Comment by thomasaarholt** [comment]
Works much better!

**Comment by francisco-dlp** [comment]
Fixed in #1544
